opacus/grad_sample/conv.py

```diff
@@ -1,16 +1,52 @@
 #!/usr/bin/env python3
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 
-from typing import Dict, Union
+from typing import Dict, Union, Tuple
 
 import numpy as np
+import time
 import torch
 import torch.nn as nn
+import torch.nn.functional as F
 from opacus.utils.tensor_utils import unfold3d
 
 from .utils import register_grad_sampler
 
 
+def unfold2d(
+    input,
+    kernel_size: Tuple[int, int],
+    padding: Tuple[int, int],
+    stride: Tuple[int, int],
+    dilation: Tuple[int, int],
+):
+    # print("input shape", input.shape)
+    *shape, H, W = input.shape
```

Comment:
*shape?

Reply:
Apparently this puts in `shape` all elements but the last two. Nice, didn't know this syntax!
```python
>>> *shape, H, W = (1, 2, 3, 4, 5)
shape = [1, 2, 3]
```

Reply:
TIL as well!